src/main/java/io/wispforest/lavender/structure/BlockStatePredicate.java

```diff
@@ -17,13 +18,15 @@ public interface BlockStatePredicate {
      * a full state match
      */
     BlockStatePredicate NULL_PREDICATE = new BlockStatePredicate() {
+        @NotNull
         @Override
-        public BlockState preview() {
-            return Blocks.AIR.getDefaultState();
+        public BlockState[] previewBlockstates() {
+            return new BlockState[]{Blocks.AIR.getDefaultState()};
```

Comment:
Don't allocate a new array every time this function is called, store it in a private field